scipy/integrate/tests/test_quadrature.py

```diff
@@ -538,6 +542,13 @@ def func(x):
 
 
 def cumulative_simpson_nd_reference(y, *, x=None, dx=None, initial=None, axis=-1):
+    # Use cumulative_trapezoid if length of y < 3
+    if y.shape[axis] < 3:
+        if initial is None:
+            return cumulative_trapezoid(y, x=x, dx=dx, axis=axis, initial=None)
+        else:
+            return initial + cumulative_trapezoid(y, x=x, dx=dx, axis=axis, initial=0)
```

Comment:
Hmm I'll have to investigate why this branching is necessary.

Reply:
@nprav I'm confused - it looks like you deprecated the `initial` behavior for `cumulative_trapezoid` for input other than `0` or `None`, yet included support for other values of `initial` in `cumulative_simpson`. Why would it be OK for `cumulative_simpson` if it's not OK for `cumulative_trapzoid`? We wouldn't want to release `cumulative_simpson` with this behavior and then deprecate it.

Reply:
Correct, ideally after the deprecation period we would add back support for initial in cumulative_trapezoid as it is implemented for cumulative_simpson. Is there past precedence for changing API behaviour in this way?

Reply:
Yes, there is precedent. The thing that comes to mind is https://github.com/scipy/scipy/pull/6100/files, which deprecated using `stats.mannwhitneyu` without specifying `alternative`, then at the end of the deprecation period the default behavior changed and then function could be called without `alternative` again. I am not sure that everyone loves this sort of plan, but I don't know of a better way.